src/PersistStore.ts

```diff
@@ -159,11 +163,17 @@ export class PersistStore<T, P extends keyof T> {
             let propertyData = property.serialize(target[property.key]);
 
             if (propertyData instanceof ObservableMap) {
-              const mapArray: any = [];
-              propertyData.forEach((v, k) => {
+              const mapArray: any[] = [];
+              propertyData.forEach((v: any, k: any) => {
                 mapArray.push([k, toJS(v)]);
               });
               propertyData = mapArray;
+            } else if (propertyData instanceof ObservableSet) {
+              const setArray: any[] = [];
+              propertyData.forEach((v: any) => {
+                setArray.push(toJS(v));
+              });
+              propertyData = setArray;
```

Comment:
Would this be better or confusing?

```suggestion
              propertyData = propertyData.map(toJS);
```

Reply:
Thank you, agreed